### PR TITLE
Add metrics binary generator

### PR DESCRIPTION
**Pull request by puzpuzpuz**
* Adds metrics binary generator to be used for testing of non-Java clients
* Also includes minor code style improvements

Corresponding Node.js client PR: https://github.com/hazelcast/hazelcast-nodejs-client/pull/751

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/impl/MetricsCompatibilityFileGenerator.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        MetricDescriptor metric2 = metric1.copy()
+                                          .withMetric("deltaMetric2");
+        compressor.addDouble(metric2, -4.2);
+
+        MetricDescriptor metric3 = supplier.get()
+                                           .withMetric("differentMetric1")
+                                           .withUnit(BYTES);
+        compressor.addLong(metric3, Integer.MAX_VALUE);
+
+        return compressor.getBlobAndReset();
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/internal/metrics/impl/MetricsCompatibilityFileGenerator.java:74]: I would add a long name with length between 128 and 255 chars to make sure the non-java clients produce the same output (we currently use an unsigned byte for the length of the text).

Comment (puzpuzpuz) [reply hazelcast/src/test/java/com/hazelcast/internal/metrics/impl/MetricsCompatibilityFileGenerator.java:74]: Good point. Addressed in 125b14fe8e80ddc038138ef147482f186645d86c

**Review by blazember — APPROVED** [review]
LGTM :+1:

**Comment by puzpuzpuz** [comment]
@blazember thanks for the review.

**Review by mdumandag — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@mdumandag thanks for the review!

**Pull request merged**